src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -476,6 +482,19 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                             }
                         }
                     }
+                 
+                    // Determinate if we need to use another primary group for container user
```

Comment:
Could you provide more context in the code comments regarding why this check is needed?